### ISSUE TITLE
Unexpected "JRE container" entry derived from pom.xml for Java SE 13

### ISSUE THREAD

**Issue by anthonyvdotbe** [redhat-developer/vscode-java#951]
When opening a Maven project which uses maven.compiler.source/target 13, the derived `.classpath` file uses `J2SE-1.5` as its JRE container, instead of the expected JavaSE-12.

In other words: if the value for the Maven compiler properties is unsupported/unrecognized, it should default to the highest currently-supported version. This is much more likely to be what the user wants/expects. For example, with the new JDK release cadence (the first JDK 14 build is already released), it's more common (for me at least) to try out early-access JDK builds.

##### Environment
- Operating System: Windows 10 Pro 1809 (17763.557)
- JDK version: 12.0.1
- Visual Studio Code version: 1.35.1
- Java extension version: 0.46.0

##### Steps To Reproduce
1. unzip and open [sample-maven-project.zip](https://github.com/redhat-developer/vscode-java/files/3293950/sample-maven-project.zip)
2. let the extension generate the project files

##### Current Result
The .classpath file contains the following:

```
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/J2SE-1.5">
		<attributes>
			<attribute name="maven.pomderived" value="true"/>
		</attributes>
	</classpathentry>
```

##### Expected Result
The .classpath file contains the following:

```
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-12">
		<attributes>
			<attribute name="maven.pomderived" value="true"/>
		</attributes>
	</classpathentry>
```

**Comment by fbricon** [comment]
upstream issue: https://bugs.eclipse.org/bugs/show_bug.cgi?id=549312

**Comment by fbricon** [comment]
The fixed m2e version is now consumed by jdt.ls (https://github.com/eclipse/eclipse.jdt.ls/commit/c966395a263bf6053be110b313b5bd00d246508a)

**Comment by EricZhangzf** [comment]
Test this issue in my local machine, No matter what my local jdk version is(1.8, 11, 12), The .classpath file always:
```
<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-12">
      <attributes>
            <attribute name="maven.pomderived" value="true"/>
      </attributes>
</classpathentry>
```

**Comment by fbricon** [comment]
Ok, I think I know what the issue is. We select the highest execution environment, but there might not be an actual jdk backing it.

**Comment by wolfch-elsevier** [comment]
I have the following JDKs installed:
```
/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home
/Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home
/opt/jdk-13.0.1.jdk/Contents/Home
/opt/jdk-17.0.2.jdk/Contents/Home
```
...and in Eclipse, for "Installed JREs", I have three of the above (not 13) added to base Eclipse.  With Java-11 checked off as "default".  I have the same issue as the original poster, except, Eclipse keeps reseting the `Java Build Path` in the `.classpath` from `Java-11` to `Java-1.8` :
```xml
-       <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-11">
+       <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8">
                <attributes>
-                       <attribute name="module" value="true"/>
+                       <attribute name="maven.pomderived" value="true"/>
                </attributes>
        </classpathentry>
```
This is super frustrating because due to code-generation complexity, I want to check-in the `.project`, `.classpath` and (most importantly) the `.factorypath` but obviously not the `.setttings` directory hierarchy.  

So when we import a Maven project, why can't Eclipse just honor existing Eclipse project files and not change everything?  Otherwise, I have to document a long and tedious manual reconfiguration process, with screen shots, in particular `Annotation Processing` ==> `Factory Path`, which I was hoping to pre-can by checking in the `.factorypath` which looks like:
```xml
<factorypath>
    <factorypathentry kind="PLUGIN" id="org.eclipse.jst.ws.annotations.core" enabled="true" runInBatchMode="false"/>
    <factorypathentry kind="EXTJAR" id="/Users/ma-wolf2/.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar" enabled="true" runInBatchMode="false"/>
    <factorypathentry kind="EXTJAR" id="/Users/ma-wolf2/.m2/repository/com/google/guava/guava/23.0/guava-23.0.jar" enabled="true" runInBatchMode="false"/>
    <factorypathentry kind="EXTJAR" id="/Users/ma-wolf2/.m2/repository/com/google/auto/service/auto-service/1.0-rc6/auto-service-1.0-rc6.jar" enabled="true" runInBatchMode="false"/>
    <factorypathentry kind="EXTJAR" id="/Users/ma-wolf2/.m2/repository/com/google/auto/value/auto-value/1.7.1/auto-value-1.7.1.jar" enabled="true" runInBatchMode="false"/>
</factorypath>
```
In summary, I have the same issue as OP, but in my case, Eclipse resets and downgrades from `Java-11` to `Java-1.8` in the `.classpath` but in addition, I want to supply a `.factorypath` from source control.
